app/models/spree/avalara_transaction.rb

```diff
@@ -65,8 +94,8 @@ def cancel_order_to_avalara(doc_type = 'SalesInvoice')
 
       AVALARA_TRANSACTION_LOGGER.debug cancel_tax_result
 
-      if cancel_tax_result == 'error in Tax'
-        return 'Error in Tax'
+      if cancel_tax_result == 'Error in Cancel Tax'
```

Comment:
It's not a good idea to compare with strings. Better to use symbols.

The overall approach with comparing for the error name, and returning the same error name is quite inappropriate. There must be something better to do, like have a custom error class, and those would be the custom named exceptions.

See that as an example:

https://github.com/boomerdigital/musicmagpie/blob/master/lib/magpie/base.rb#L5-L6